acceptancetests/assess_resolve.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+""" Ensure resolve command resolves hook issues.
+
+This test covers the resolve command (and the option --no-retry)
+
+The test:
+  - Uses a simple custom charm that instruments failure in the install hook
+  - Ensures that 'resolve' retries the failed install hook and that the
+    followup 'start' hook is fired.
+  - Ensures that 'resolve --no-retry' does NOT re-run the failed install hook
+    and instead goes ahead and runs the follow up 'started' hook.
+"""
+
+from __future__ import print_function
+
+import argparse
+import logging
+import sys
+
+from deploy_stack import (
+    BootstrapManager,
+    )
+from jujupy.models import (
+    temporary_model,
+)
+from jujupy.wait_condition import (
+    UnitInstallCondition,
+)
+from jujucharm import local_charm_path
+from utility import (
+    add_basic_testing_arguments,
+    configure_logging,
+    )
+
+
+__metaclass__ = type
+
+
+log = logging.getLogger('assess_resolve')
+
+
+class ResolveCharmMessage:
+    """Workload active message differs if the install hook completed."""
+    ACTIVE_NO_INSTALL_HOOK = 'No install hook'
+    ACTIVE_INSTALL_HOOK = 'Install hook succeeded'
+    INSTALL_FAIL = 'hook failed: "install"'
+
+
+def assess_resolve(client, local_charm='simple-resolve'):
+    local_resolve_charm = local_charm_path(
+        charm=local_charm, juju_ver=client.version)
+    ensure_retry_does_not_rerun_failed_hook(client, local_resolve_charm)
+
+
+def ensure_retry_does_not_rerun_failed_hook(client, resolve_charm):
+        with temporary_model(client, "no-retry") as temp_client:
+            unit_name = 'simple-resolve/0'
+            temp_client.deploy(resolve_charm)
+            temp_client.wait_for(UnitInstallError(unit_name))
+            temp_client.juju('resolve', ('--no-retry', unit_name))
+            temp_client.wait_for(
```

Comment:
We need to also check show-status-log to ensure the expected hooks have been run,
eg I can't see where we check that the start hook fires when the install hook failure is resolved

Reply:
I missed the new charm in the original push, it's there now. The charm updates the message in the install and start hook script. The UnitInstallActive will wait until it sees the message "'No install hook'" (i.e. start hook fired, the install hook never resolved.)

We could check the show-status-log for the string "juju-unit  executing    running start hook" if we feel we need further confirmation that it ran.

Reply:
I think it's fine, Seeing the charm makes sense. Maybe a code comment saying that the charm sets status etc etc